app/code/Magento/MediaGallery/Plugin/Wysiwyg/Images/Storage.php

```diff
@@ -92,4 +101,36 @@ public function afterDeleteFile(StorageSubject $subject, StorageSubject $result,
 
         return $result;
     }
+
+    /**
+     * Delete media data after the folder delete action from Wysiwyg
+     *
+     * @param StorageSubject $subject
+     * @param mixed $result
+     * @param string $path
+     *
+     * @return null
+     *
+     * @throws CouldNotDeleteException
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterDeleteDirectory(StorageSubject $subject, $result, $path)
+    {
+        if (!is_string($path)) {
+            return $result;
+        }
+
+        try {
+            $mediaDirectoryRead = $this->filesystem->getDirectoryRead(DirectoryList::MEDIA);
+            $relativePath = $mediaDirectoryRead->getRelativePath($path);
+            if ($mediaDirectoryRead->isExist($relativePath) === false) {
```

Comment:
Is this check necessary? Considering that it is `afterDeleteDirectory` method body?

Reply:
Hi @sivaschenko. 

The idea was the following. If the original method's (`deleteDirectory`) `$path` for some reason does not exist, we most likely will not have an exception, just successful method execution (see: https://github.com/magento/magento2/blob/3adb1eb96d6427e4d4c91aad65eac38aea77b8f5/lib/internal/Magento/Framework/Filesystem/Directory/Write.php#L182). 
After that our plugin will execute the DB query which does not make sense. 

This is an edge-case. If you believe that we don't need this check, I'm OK as well. 

Thank you!

Reply:
@rogyar I think this check should be removed as the plugin is `afterDeleteDirectory` so the directory should not exist when it's executed anyway